src/software/ai/hl/stp/tactic/tactic_visitor.h

```diff
@@ -0,0 +1,148 @@
+#pragma once
+
+// We forward-declare all the tactics because if we include them we induce a
+// circular dependency between the Individual library for each tactic and this
+// visitor. Ex: `CherryPickTactic` includes `TacticVisitor`, but `TacticVisitor`
+// also includes `CherryPickTactic`
+class CherryPickTactic;
```

Comment:
no because all_tactics is included by move_rule_manager only. This is because if tactic_visitor included all_tactics and then each tactic included tactic visitor, we'd get a circular dependency